src/main/java/duke/Command.java

```diff
@@ -0,0 +1,98 @@
+package duke;
+
+/**
+ * Represents the action requested by the user that Duke understands.
+ */
+public class Command {
+    private String operation;
+    private int index;
+    private String description;
+    private String time;
+
+    /**
+     * Class constructor specifying an operation that requires no other input.
+     *
+     * @param operation the operation for Duke to execute
+     */
+    public Command(String operation) {
+        this.operation = operation;
+    }
+
+    /**
+     * Class constructor specifying an operation that takes a task index.
+     *
+     * @param operation The operation for Duke to execute
+     * @param index The index of the task in Duke's task list
+     */
+    public Command(String operation, int index) {
+        this.operation = operation;
+        this.index = index;
+    }
+
+    /**
+     * Class constructor specifying an operation that takes a description.
+     *
+     * @param operation The operation for Duke to execute
+     * @param description Description of the task
+     */
+    public Command(String operation, String description) {
+        this.operation = operation;
+        this.description = description;
+    }
+
+    /**
+     * Class constructor specifying an operation that takes a description and
+     * a time string. The time string will be converted to a LocalDate object
+     * if it is of the form YYYY-MM-DD.
+     *
+     * @param operation The operation for Duke to execute
+     * @param description Description of the task
+     * @param time Time that the task will occur / should be done by
+     */
+    public Command(String operation, String description, String time) {
+        this.operation = operation;
+        this.description = description;
+        this.time = time;
+    }
+
+    /**
+     * Returns the specific operation as a String that Duke is able to understand.
+     *
+     * @return Duke operation
+     */
+    public String getOperation() {
+        return operation;
+    }
```

Comment:
I like how you used getters to get information instead of accessing it directly from the instance itself!